backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/utils/MDevTypesUtils.java

```diff
@@ -30,4 +41,36 @@ public static boolean isMdevDisplayOnSupported(Version version) {
         // See https://bugzilla.redhat.com/show_bug.cgi?id=1633623 for details and discussion
         return version.greaterOrEquals(Version.v4_3);
     }
+
+    public static List<VmDevice> convertDeprecatedCustomPropertyToVmDevices(String property, Guid vmId) {
+        if (property.isEmpty()) {
+            return Collections.emptyList();
+        }
+        try {
+            List<VmDevice> devices = new ArrayList<>();
+            Boolean nodisplay = Boolean.FALSE;
+            for (String type : property.split(",")) {
+                if (type.equals(NODISPLAY)) {
+                    nodisplay = Boolean.TRUE;
+                } else {
+                    VmDevice device = new VmDevice();
+                    device.setId(new VmDeviceId(Guid.newGuid(), vmId));
+                    device.setType(VmDeviceGeneralType.MDEV);
+                    device.setDevice(VmDeviceType.VGPU.getName());
+                    device.setAddress("");
+                    device.setManaged(true);
+                    device.setPlugged(true);
+                    Map<String, Object> specParams = new HashMap<>();
+                    specParams.put(MDevTypesUtils.MDEV_TYPE, type);
+                    specParams.put(MDevTypesUtils.NODISPLAY, nodisplay);
+                    device.setSpecParams(specParams);
+                    devices.add(device);
```

Comment:
should be in VmDeviceCommonUtils?

Reply:
no, it's very specific to backward compatibility of mediated devices